File: app/src/main/java/ru/samtakoy/features/notifications/StartupReceiver.java
package ru.samtakoy.features.notifications;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

/**
 * TODO перейти на WorkManager?
 */
public class StartupReceiver extends BroadcastReceiver {


    @Override
    public void onReceive(Context context, Intent intent) {
        context.startService(
                NotificationsPlannerService.getOnBootReSchedulingIntent(context)
        );
    }
}

File: androidApp/src/main/kotlin/ru/samtakoy/oldlegacy/features/notifications/StartupReceiver.kt
package ru.samtakoy.oldlegacy.features.notifications

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

/**
 * TODO перейти на WorkManager?
 */
class StartupReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent?) {
        context.startService(
            NotificationsPlannerService.getOnBootReSchedulingIntent(context)
        )
    }
}
